lib/bracket-matcher-view.coffee

```diff
@@ -24,7 +25,9 @@ class BracketMatcherView
     @tagFinder = new TagFinder(@editor)
     @pairHighlighted = false
     @tagHighlighted = false
+    @commentOrStringSelector = SelectorCache.get('comment.* | string.*')
 
+    @subscriptions.add @editor.onDidTokenize(@updateMatch)
```

Comment:
So, this line is a fix for a special case that only happens when the following two conditions are **both** met:

1) a file is opened **or** the file grammar is changed (ie: from plain text to javascript)
2) the cursor initially starts on a commented bracket **or** a bracket in a string.

Normally, you'd get an incorrect match in this situation.

This happens because bracket matching occurs before tokenization is enabled. The editor doesn't know what's a comment or string yet, so incorrect matches are made.<sup>[1]</sup>

Adding this listener forces an update after tokenization kicks in, thus correcting the initial bad match. The listener is only triggered once, on file load or grammar change, so it's fairly efficient.

[1] Incidentally, these matches will be corrected on the first cursor move, but it shouldn't happen in the first place.